Tasks/DownloadBuildArtifactsV0/main.ts

```diff
@@ -265,68 +265,80 @@ async function main(): Promise<void> {
+                            endpointUrl: endpointUrl,
+                            projectId: projectId,
+                            buildId: buildId,
+                            handler: handler
+                        };
+
+                        const downloadHandler: DownloadHandlerContainerZip = new DownloadHandlerContainerZip(config);
+                        const downloadPromise: Promise<models.ArtifactDownloadTicket[]> = executeWithRetries(
+                            operationName,
+                            () => downloadHandler.downloadResources(),
+                            retryLimit
+                        ).catch((reason) => {
                             reject(reason);
                             return;
                         });
                         
-                        downloadPromises.push(downloadZipPromise);
-                        await downloadZipPromise;
-
+                        downloadPromises.push(downloadPromise);
+                        await downloadPromise;
                     }
                     else {
-                        let downloader = new engine.ArtifactEngine();
-
-                        console.log(tl.loc("DownloadingContainerResource", artifact.resource.data));
-                        var containerParts = artifact.resource.data.split('/');
-
-                        if (containerParts.length < 3) {
-                            throw new Error(tl.loc("FileContainerInvalidArtifactData"));
-                        }
-
-                        var containerId = parseInt(containerParts[1]);
-                        var containerPath = containerParts.slice(2, containerParts.length).join('/');
-
-                        if (containerPath == "/") {
-                            //container REST api oddity. Passing '/' as itemPath downloads the first file instead of returning the meta data about the all the files in the root level. 
-                            //This happens only if the first item is a file.
-                            containerPath = ""
-                        }
-
-                        var itemsUrl = endpointUrl + "/_apis/resources/Containers/" + containerId + "?itemPath=" + encodeURIComponent(containerPath) + "&isShallow=true&api-version=4.1-preview.4";
-                        console.log(tl.loc("DownloadArtifacts", artifact.name, itemsUrl));
-
-                        var variables = {};
-                        var webProvider = new providers.WebProvider(itemsUrl, templatePath, variables, handler);
-                        var fileSystemProvider = new providers.FilesystemProvider(downloadPath);
-
-                        downloadPromises.push(downloader.processItems(webProvider, fileSystemProvider, downloaderOptions).catch((reason) => {
+                        const operationName: string = `downloadContainer- ${artifact.name}`;
+                        const config: IContainerHandlerConfig = {
+                            artifactInfo: artifact,
```

Comment:
Can we assign common fields out of "if" statement to config, and set more specific inside?

Reply:
Fixed it. 
Thanks!